semgrep-core/src/core/Semgrep_core_response_util.ml

```diff
@@ -0,0 +1,74 @@
+(*
+   Utilities for working with the types defined in Semgrep_core_response.atd
+   (Semgrep_core_response_t module)
+*)
+
+open Semgrep_core_response_t
+
+let compare_position (a : position) b = Int.compare a.offset b.offset
+
+let compare_location (a : location) b =
+  let c = String.compare a.path a.path in
```

Comment:
There may be a bug here, it should probably be `String.compare a.path b.path`, I think that `c` is always going to be `0`. Maybe a Semgrep rule to add for OCaml?

Reply:
ohoh, good one, more reason to use deriving ord to automatically generate those.

Reply:
ah, it's like the semgrep rule `$X == $X` which I thought was completely useless. 😅

Reply:
I guess you can add $M_.compare $Y $Y in the local semgrep.yml

Reply:
added to semgrep-rules: https://github.com/returntocorp/semgrep-rules/pull/1714